### PR TITLE
Improve handling of TypeScript versions

### LINKED ISSUE #51: Test against typescript@latest
Ref: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/17626#issuecomment-312284279

### PR DESCRIPTION
**Pull request by andy-ms**
Previously we would test with the version specified by a `// TypeScript Version:` comment, and also test with a higher version if that failed.
Now we will begin by testing with `@next`, and also test with the minimum specified version. If there is an error in the minimum version, we will try all versions to find the oldest one that works (which is one higher than the newest one that breaks).
Also removes the `--tsNext` flag since that is a given now. If this is merged I will remove that from `types-publisher` too.
Also includes fixes to the `dt-header` tests which had gotten out of date.

Fixes #51 (Ref: DefinitelyTyped/DefinitelyTyped#17626)

### CODE REVIEW

File: src/rules/definitelytyped-header-parser.ts [hunk 1]
```diff
@@ -8,6 +8,9 @@ export namespace TypeScriptVersion {
 	export const all: ReadonlyArray<TypeScriptVersion> = ["2.0", "2.1", "2.2", "2.3", "2.4"];
 	/** Latest version that may be specified in a `// TypeScript Version:` header. */
 	export const latest = "2.4";
+	export function range(min: TypeScriptVersion) {
+		return all.filter(v => v >= min);
```
Comment (RyanCavanaugh) [inline src/rules/definitelytyped-header-parser.ts]: Fine as-is but let's make a mental note to panic if we release a version number like 2.13 (I don't expect this will happen)

**Comment by leonard-thieu** [comment]
Wasn't this supposed to be testing against `@latest` instead of `@next`?

**Comment by andy-ms** [comment]
@leonard-thieu We will soon be running tests nightly to ensure that `next` always works with every package. So any breaks will be dealt with by the TS team.
